src/beamlime/core/service.py

```diff
@@ -137,15 +137,18 @@ def _run_loop(self) -> None:
                 remaining = max(0.0, self._poll_interval - elapsed)
                 if remaining > 0:
                     time.sleep(remaining)
-        except RuntimeError:
+        except (RuntimeError, ValueError):
```

Comment:
Does it make sense to do a catch all here?
```suggestion
        except Exception:
```
We do want to stop the loop if we trigger *any* unexpected error.